middleware/locale/locale.go

```diff
@@ -0,0 +1,61 @@
+package locale
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/mholt/caddy/middleware"
+	"github.com/mholt/caddy/middleware/locale/method"
+)
+
+// Locale is a middleware to detect the user's locale.
+type Locale struct {
+	Next             middleware.Handler
+	AvailableLocales []string
+	Methods          []method.Method
+	PathScope        string
+	Configuration    *method.Configuration
+}
+
+// ServeHTTP implements the middleware.Handler interface.
+func (l *Locale) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error) {
+	if !middleware.Path(r.URL.Path).Matches(l.PathScope) {
+		return l.Next.ServeHTTP(w, r)
+	}
```

Comment:
There's been a very specific example on how to do scoping, in one of the comments.

And one for how a well-written ```ServeHTTP``` would look like after that:

```go
wantedByVisitor := []BrowserLocale{}
for _, reader := range l.ApplicableGetters {
  lang := reader(r, l.Conf) // reads "accept-language" from HTTP headers, Cookies…
  wantedByVisitor = append(wantedByVisitor, lang...)
}
sort(wantedByVisitor) // by precedence

match := l.reduceToAvailable(wanted)
if len(match) == 0 {
  match = append(match, l.DefaultLocale) // or return 406
}
r.Header.Set(…, match[0]) // or the whole list

return l.Next.ServeHTTP(w, r)
```